assets/js/components/data.js

```diff
@@ -356,79 +369,94 @@ const data = {
 	/**
 	 * Sets data in the cache.
 	 *
-	 * @param {object} type  The data object.
-	 * @param {string} key   The cache key,
-	 * @param {object} value The value to cache.
+	 * @param {string} key  The cache key,
+	 * @param {mixed}  data The data to cache.
 	 */
-	setCache( type, key, value ) {
-		if ( value.error || value.errors ) {
+	setCache( key, data ) {
+		if ( 'undefined' === typeof data ) {
+			return;
+		}
+
+		// Specific workaround to ensure no error responses are cached.
+		if ( data && 'object' === typeof data && ( data.error || data.errors ) ) {
 			return;
 		}
 
-		if ( window.sessionStorage ) {
-			const toStore = {
-				value,
-				date: Date.now() / 1000
-			};
-			setCache( 'sessionStorage', type + '::' + key, JSON.stringify( toStore ) );
+		lazilySetupLocalCache();
+
+		googlesitekit.admin.datacache[ key ] = data;
+
+		if ( ! window.sessionStorage ) {
+			return;
 		}
+
+		const toStore = {
+			value: data,
+			date: Date.now() / 1000
+		};
+		window.sessionStorage.setItem( 'googlesitekit_' + key, JSON.stringify( toStore ) );
 	},
 
 	/**
 	 * Gets data from the cache.
 	 *
-	 * @param {object} type   The data object.
 	 * @param {string} key    The cache key,
 	 * @param {int}    maxAge The cache TTL in seconds.
 	 *
 	 * @return {mixed} Cached data, or false if lookup failed.
 	 */
-	getCache( type, key, maxAge, hashlessKey = '' ) {
+	getCache( key, maxAge ) {
 
 		// Skip if js caching is disabled.
 		if ( googlesitekit.admin.nojscache ) {
 			return false;
 		}
 
-		// Require sessionStorage for local caching.
+		lazilySetupLocalCache();
+
+		// Check variable cache first.
+		if ( 'undefined' !== typeof googlesitekit.admin.datacache[ key ] ) {
+			return googlesitekit.admin.datacache[ key ];
+		}
+
 		if ( ! window.sessionStorage ) {
 			return false;
 		}
 
-		const cache = JSON.parse( window.sessionStorage.getItem( type + '::' + key ) );
-		const datacache = googlesitekit.admin.datacache && JSON.parse( googlesitekit.admin.datacache );
+		// Check persistent cache.
+		const cache = JSON.parse( window.sessionStorage.getItem( 'googlesitekit_' + key ) );
+		if ( cache && 'object' === typeof cache && cache.date ) {
 
-		// Check the datacache, which only includes fresh data at load.
-		if ( ! cache && datacache && datacache[ 'googlesitekit_' + key ] ) {
-			return datacache[ 'googlesitekit_' + key ];
-		}
+			// Only return value if no maximum age given or value is newer than it.
```

Comment:
This wording is slightly confusing. Perhaps:
"Only return value if no maximum age given or if cache age is less than the maximum."